### DISCUSSION TITLE
Time in ROI using Deeplabcut

### DISCUSSION THREAD

**Discussion by paulagsotres** [bonsai-rx/bonsai#1192] [Q&A] [answered]
Hello!

My name is Paula, I am a PhD student from Bordeaux that is starting to use Bonsai. What I am trying to do is to use the GetBodyPart node to track the nose of the mouse with my pretrained DLC model, and count the time that that body part is in a specific ROI (i.e. around the objects of the ends of an L maze). This tracking is more refined than the centroid for my experiment. However, I can't manage to do this! Any help?

Thank  you so much in advance!

I attach the workflow, and a screenshot of my experimental set up. 
![imagen](https://user-images.githubusercontent.com/120389016/207056968-a129f4c6-1281-4ff9-b2b2-da10e5e3bc3e.png)

**Comment by bruno-f-cruz** [comment, marked as answer]
Check the attached workflow for a possible solution.
Basically, from a frame extract whatever `Point2f` you need (in your case the position of the `BodyPart`). Then, define a `RegionIfInterest` in the image and pair it up with each point previously generated. 
Finally use the `ContainsPoint` operator that takes a `Tuple<Rect, Point2f>` and returns whether the point is inside the specified region or not. 
You can then simply count the number of frames that this condition evaluates to True and multiply it by your frame rate to get time spent in the selected area.

Hope it helps,
Bruno
[ROI_POINT_detection.zip](https://github.com/bonsai-rx/bonsai/files/10208723/ROI_POINT_detection.zip)

**Comment by paulagsotres** [comment]
Thank you so much for your fast answer!

I think I am missing something because I don't understand how can I create a CreatePoint2f with the dynamic coordinates that GetBodyPart gives me...Can I link the coordinates from GetBodyPart somehow by externalizing them to the coordinates that form the Point2f?

Sorry if these are naive questions!

**Reply by bruno-f-cruz** [reply]
Right click on the GetBodyPart and extract position 
![image](https://user-images.githubusercontent.com/7049351/207103459-081e1137-87ff-4151-adc4-feed6b30a60c.png)

**Reply by paulagsotres** [reply]
Thank so much Bruno! It works perfectly!
This is an amazing commmunity 👍

**Comment by kapilsaxena33** [comment]
Hi Bruno, Is there a way to count the number of entries in the ROI as well in the above workflow and timestamp them (or Frame Number with  0/1)?

**Reply by bruno-f-cruz** [reply]
You can add a `DistinctUntilChanged -> Condition -> Timestamp` after the `ContainsPoint`.

**Reply by kapilsaxena33** [reply]
Thanks Bruno, it worked.
